### ISSUE TITLE
The `padding` type should be set as Scaler for torch.nested.to_padded_tensor (currently set to double)

### ISSUE THREAD

**Issue by sumitbinnani** [pytorch/pytorch#176625]
_Apologies to create a new issue. I didn't have permission to reopen  #175866_

## Original Bug Report: #175866
Padding value loses precisions for long nest tensor when using `to_padded_tensor`. The bug has been independently verified by multiple people. Minimal example to reproduce:
```python
import torch
from torch import tensor

sample = torch.nested.nested_tensor([tensor([1, -7704671799122851728]), tensor([-7704671799122851728])])
padded_tensor = torch.nested.to_padded_tensor(sample, padding=-7704671799122851728)
expected_padded_tensor = tensor([[ 1, -7704671799122851728], [-7704671799122851728, -7704671799122851728]])
torch.equal(padded_tensor, expected_padded_tensor) # -> False!!
```

## Why create new? 
The original issue was closed by https://github.com/pytorch/pytorch/issues/175866#issuecomment-3985385891
> the number is not representable as cpp double, so not much we can do here 

I believe we have a path to address this. The issue arises because the `padding` value is cast through a C++ double i.e. **the current code casts the long -> double -> long**. I believe can be fixed by **can be fixed by setting `padding` type to `Scalar`**.

## Hypothesis Check
I have a put a [PR](https://github.com/pytorch/pytorch/pull/176029) that shows how by correctly setting the `padding` type to `Scalar` will resolve the issue. The code passes old test cases and I have added new one for testing the new change.

Thank you for you attention and a second look at a previously closed issue.

cc: @jcaip @soulitzer @malfet 

cc @cpuhrsch @jbschlosser @bhosmer @drisspg @soulitzer @davidberard98 @YuqingJ
